pkg/controller/petset/identity_mappers.go

```diff
@@ -0,0 +1,202 @@
+package petset
+
+import (
+	"crypto/md5"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	api_pod "k8s.io/kubernetes/pkg/api/pod"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+const svcLocalSuffix = "svc.cluster.local"
+
+// identityMapper is an interface for assigning identities to a pet.
+// All existing identity mappers just append "-(index)" to the petset name to
+// generate a unique identity. This is used in claims/DNS/hostname/petname
+// etc. There's a more elegant way to achieve this mapping, but we're
+// taking the simplest route till we have data on weather users will need
+// more customization.
+type identityMapper interface {
+	AddIdentity(id string, pet *api.Pod)
```

Comment:
actually it's clearer as pet IMO, because you only assign identities to pets. All pets are pods but not the other way, and we don't have a "pet" type so we have to rely on documentation to bring out the difference in code.